client-ui/src/app/model/operation/operation-property.models.ts

```diff
@@ -42,77 +42,61 @@ export namespace OperationProperty {
-    return propertyValue._typeHint && (propertyValue._typeHint as string).startsWith('expr.')
-  }
-
   export function parseExtraOptions(nativeProperties: NativeProperties, attributesList: AttributeVM[]): ExtraProperties {
-    const extraProperties: ExtraProperties = {
-      primitive: [],
-      expression: [],
-      json: []
-    }
-
-    return Object.keys(nativeProperties)
+    return Object.entries(nativeProperties)
+      .filter(([, v]) => v != null)
       .reduce(
-        (result, key) => {
-          const value = nativeProperties[key]
-
-          if (value === undefined || value === null) {
-            return result
+        (acc, [key, value]) => {
+          if (typeof value === 'string' || typeof value === 'number' || typeof value === 'boolean') {
+            const primitiveValue: ExtraPropertyValuePrimitive = {
+              label: key,
+              value: value
+            }
+            return {...acc, primitive: [...acc.primitive, primitiveValue]}
           }
-
-          const primitivePropValue = {
-            label: key,
-            value
+          else if (typeof value === 'object' && isExpresionProperty(value)) {
+            return {...acc, expression: [...acc.expression, expressionValue(key, [value], attributesList)]}
           }
-
-          if (typeof value === 'string' || typeof value === 'number') {
-            extraProperties.primitive.push(primitivePropValue)
+          else if (typeof Array.isArray(value) && value.length && isExpresionProperty(value[0])) {
+            return {...acc, expression: [...acc.expression, expressionValue(key, value, attributesList)]}
           }
-          else if (typeof value === 'object' && isExpresionProperty(value)) {
-            try {
-              const decoratedValue = decorateExpressionProperty(primitivePropValue, attributesList)
-              extraProperties.expression.push(decoratedValue)
-            } catch (e) {
-              console.error(`Expresion decoration error`, e)
+          else {
+            const jsonValue: ExtraPropertyValueJson = {
+              label: humanizeCase(key),
+              value
             }
+            return {...acc, json: [...acc.json, jsonValue]}
           }
-          else if (Array.isArray(value) || typeof value === 'object') {
-            extraProperties.json.push(
-              decorateJsonProperty(primitivePropValue)
-            )
-          }
-
-          return result
         },
-        extraProperties
+        {
+          primitive: [],
+          expression: [],
+          json: []
+        }
       )
+  }
 
+  function expressionValue(key: string, exprs: Record<string, any>[], attrs: AttributeVM[]): ExtraPropertyValueExpression {
+    return {
+      label: humanizeCase(key),
+      value: exprs.map(e => ({
+        value: getText(e, attrs),
+        rawValue: e
+      }))
+    }
   }
 
+  function humanizeCase(str: string): string {
```

Comment:
=> toHumanizeCase

Reply:
"humanize" is a verb

Reply:
Oh, you are right. In that case it is ok.